plugins/modules/purefa_info.py

```diff
@@ -2572,16 +2572,15 @@ def generate_certs_dict(array):
 
 def generate_kmip_dict(array):
     kmip_info = {}
-    api_version = array._list_available_rest_versions()
-    if P53_API_VERSION in api_version:
-        kmips = array.list_kmip()
-        for kmip in range(0, len(kmips)):
-            key = kmips[kmip]["name"]
-            kmip_info[key] = {
-                "certificate": kmips[kmip]["certificate"],
-                "ca_cert_configured": kmips[kmip]["ca_certificate_configured"],
-                "uri": kmips[kmip]["uri"],
-            }
+    kmips = list(array.get_kmip().items)
+    for kmip in range(0, len(kmips)):
+        key = kmips[kmip].name
+        kmip_info[key] = {
+            "certificate": kmips[kmip].certificate.name,
+            "ca_certificate": kmips[kmip].ca_certificate,
```

Comment:
Few scenarios not having ca_certificate field.
This modification works - "ca_certificate": getattr(kmips[kmip], 'ca_certificate', None),